cdap-app-fabric/src/main/java/org/apache/hadoop/fs/ftp/FTPInputStream.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.hadoop.fs.ftp;
+
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.hadoop.classification.InterfaceAudience;
+import org.apache.hadoop.classification.InterfaceStability;
+import org.apache.hadoop.fs.FSInputStream;
+import org.apache.hadoop.fs.FileSystem;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * FTP Input Stream, copied from Hadoop and modified, that doesn't thrown an exception when seeks are attempted to the
```

Comment:
add link or fully qualified class name of the copied class. also typo: `doesn't throw an exception`